src/lib.rs

```diff
@@ -226,6 +247,65 @@ pub fn read_accessions(file: &PathBuf) -> Vec<String> {
         .collect()
 }
 
+pub fn print_csv(runs: Vec<Run>) -> Result<(), std::io::Error> {
+    let mut wtr = csv::Writer::from_writer(io::stdout());
+    for run in runs {
+        wtr.write_record(&["accession", "url", "md5", "bytes"])?;
+        for read in run.reads {
+            wtr.write_record(&[&run.accession, &read.url, &read.md5, &read.bytes.to_string()])?;
+        }
+    }
+    wtr.flush()?;
+    Ok(())
+}
+
+pub fn print_csv_long(runs: Vec<Run>, keep_single_end: bool) -> Result<(), std::io::Error> {
+    let mut wtr = csv::Writer::from_writer(io::stdout());
+    match keep_single_end {
+        true => {
+            wtr.write_record(&["accession", "read1_url", "read2_url", "read3_url", "read1_md5", "read2_md5", "read3_md5", "read1_bytes", "read2_bytes", "read3_bytes"])?;
+            for run in runs {
+                let l = run.reads.len();
+                match l {
+                    1 => {
+                        wtr.write_record(&[&run.accession, &run.reads[0].url, "", "", &run.reads[0].md5, "", "", &run.reads[0].bytes.to_string(), "", ""])?;
+                    },
+                    2 => {
+                        wtr.write_record(&[&run.accession, &run.reads[0].url, &run.reads[1].url, "", &run.reads[0].md5, &run.reads[1].md5, "", &run.reads[0].bytes.to_string(), &run.reads[1].bytes.to_string(), ""])?;
+                    },
+                    3 => {
```

Comment:
I am wondering if we need the outer match. Could we instead have:

 2 || ( 3 && !keep_single_end) => {},
3 && keep_single_end => {}

Reply:
That way we can eliminate a fair bit of code.